geomag/world_magnetic_model.py

```diff
@@ -0,0 +1,294 @@
+from __future__ import division
+
+import math
+import os
+from datetime import date
+
+from .scalar_potential import scalar_potential, schmidt_quasi_normilisation, recursion_constants, _gen_2d_array
+from .latlon import LatLon
+
+
+def _gen_square_array(size_x, default=None):
+    '''Creates a square array with x by x elements'''
+    return [[default] * size_x for _ in range(size_x)]
+
+
+def _calculate_decimal_year(date_of_year):
+    '''
+    .total_seconds() call makes the function 2/3 compliant. timedelta division was added in 3.2.
+    This does limit the module to 2.7 as the minimum supported
+    '''
+    year = date_of_year.year
+    start_of_this_year = date(year, 1, 1)
+    days_this_year = (date(year + 1, 1, 1) - start_of_this_year).total_seconds()
+    days_into_year = (date_of_year - start_of_this_year).total_seconds()
+    return year + days_into_year / days_this_year
+
+
+def _convert_to_km(value, unit):
+    conversion_factor = {'ft': 3280.8399, 'm': 1000, 'km': 1}
+    try:
+        value_in_km = value / conversion_factor[unit]
+    except KeyError:
+        raise KeyError('Unknown unit: {unit}')
+    return value_in_km
+
+DEFAULT_PATH = os.path.join(os.path.dirname(__file__), 'model_data/WMM.COF')
+
+class MageneticModelData:
+    ''' Class to hold the data and calculations from the file.'''
+    _last_calculated_datetime = None
+    max_order = degree_of_expansion = 12
+    array_size = max_order + 1
+    coeff={}
+    coefficient = _gen_square_array(array_size, 0.0)
+    coefficient_dot = _gen_square_array(array_size, 0.0)
+    time_adjusted_coefficients_cache = _gen_square_array(array_size, 0.0)
+    
+    def __init__(self, file):
+        with open(file) as world_magnetic_model_file:
```

Comment:
I usually anticipate the possible errors, but here, it's true that it will raise an exception... Good enough I guess.